### PR TITLE
Complete the documentation about sharing

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: docs/sharing-design.md [hunk 1]
```diff
@@ -230,6 +230,58 @@ In fact, it is used to acknowledge the writing and is helpful for conflict
...
+revision is then sent to Bob: Bob's Cozy accepts the new revision even if it
+has no effect on the file (it was already the good name), just to resolve the
+conflict.
+
+#### Example 2
+
+![A difficult conflict](diagrams/files-conflicts-2.png)
+
+Like in the last example, Alice uploads a file and share a directory to Bob
+with this file, Bob acccepts. But the, several actions are made on the file in
```
Comment (Gara64) [inline docs/sharing-design.md]: But the -> But then?

File: docs/workers.md [hunk 2]
```diff
@@ -203,3 +203,22 @@ permission. It is done by adding this to the manifest:
...
+### Share-track
+
+The message is composed of 3 fields: the sharing ID, the rule index, and the
+doctype. The event is similar to a realtime event: a verb, a document, and
+optionaly the old version of this document.
+
+### Share-replicate and share-upload
+
+The message is composed of a sharing ID and a count of the number of errors
+(ie the number of times this job was retried).
```
Comment (Gara64) [inline docs/workers.md]: ie -> i.e.

File: docs/diagrams/replicator.ditaa [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+|      |
+|      +<-----------------------------------
+|Swift |
+|      +----------------------------------->                          upload                                                                 /------\
+|      |                                         -------------------------------------------------->                                         |      |
+\------/                                                                                                 ----------------------------------->+      |
+                                                                                                                                             | Swift|
+                                                                                                         <-----------------------------------+      |
+                                                 <--------------------------------------------------                                         |      |
+                                                                                                                                             \------/
```
Comment (Gara64) [inline docs/diagrams/replicator.ditaa]: On the generated png, the Shared and Shared+files blocks appear as databases on Alice, but not on Bob. Neither are foos, on both sides.

Comment (Gara64) [reply docs/diagrams/replicator.ditaa]: On a side note, I find the png a bit difficult to read, the formatting is sometimes a bit weird/ugly

**Comment by edas** [comment]
👌 Thank you

**Pull request merged**